translator/trace/zipkin/zipkinv2_to_traces.go

```diff
@@ -0,0 +1,454 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package zipkin
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"sort"
+	"strconv"
+	"strings"
+
+	zipkinmodel "github.com/openzipkin/zipkin-go/model"
+
+	"go.opentelemetry.io/collector/consumer/pdata"
+	otlptrace "go.opentelemetry.io/collector/internal/data/opentelemetry-proto-gen/trace/v1"
+	"go.opentelemetry.io/collector/translator/conventions"
+	tracetranslator "go.opentelemetry.io/collector/translator/trace"
+)
+
+var floatRegex, _ = regexp.Compile(`^-?\d+\.\d+$`)
+
+var intAttributes = getIntAttributes()
+
+func getIntAttributes() map[string]struct{} {
+	attrs := make(map[string]struct{})
+	attrs[conventions.AttributeNetHostPort] = struct{}{}
+	attrs[conventions.AttributeNetPeerPort] = struct{}{}
+	attrs[conventions.AttributeHTTPHostPort] = struct{}{}
+	attrs[conventions.AttributeHTTPFlavor] = struct{}{}
+	attrs[conventions.AttributeHTTPRequestContentLength] = struct{}{}
+	attrs[conventions.AttributeHTTPRequestContentLengthUncompressed] = struct{}{}
+	attrs[conventions.AttributeHTTPResponseContentLength] = struct{}{}
+	attrs[conventions.AttributeHTTPResponseContentLengthUncompressed] = struct{}{}
+	attrs[conventions.AttributeMessageID] = struct{}{}
+	attrs[conventions.AttributeMessageCompressedSize] = struct{}{}
+	attrs[conventions.AttributeMessageUncompressedSize] = struct{}{}
+	attrs[conventions.AttributeMessagingConversationID] = struct{}{}
+	attrs[conventions.AttributeMessagingPayloadSize] = struct{}{}
+	attrs[conventions.AttributeMessagingPayloadCompressedSize] = struct{}{}
+	return attrs
+}
+
+var nonSpanAttributes = getNonSpanAttributes()
+
+func getNonSpanAttributes() map[string]struct{} {
+	attrs := make(map[string]struct{})
+	for _, key := range conventions.GetResourceSemanticConventionAttributeNames() {
+		attrs[key] = struct{}{}
+	}
+	attrs[tracetranslator.TagServiceNameSource] = struct{}{}
+	attrs[tracetranslator.TagInstrumentationName] = struct{}{}
+	attrs[tracetranslator.TagInstrumentationVersion] = struct{}{}
+	attrs[conventions.OCAttributeProcessStartTime] = struct{}{}
+	attrs[conventions.OCAttributeExporterVersion] = struct{}{}
+	attrs[conventions.OCAttributeProcessID] = struct{}{}
+	attrs[conventions.OCAttributeResourceType] = struct{}{}
+	return attrs
+}
+
+// Custom Sort on
+type byOTLPTypes []*zipkinmodel.SpanModel
+
+func (b byOTLPTypes) Len() int {
+	return len(b)
+}
+
+func (b byOTLPTypes) Less(i, j int) bool {
+	diff := strings.Compare(extractLocalServiceName(b[i]), extractLocalServiceName(b[j]))
+	if diff != 0 {
+		return diff <= 0
+	}
+	diff = strings.Compare(extractInstrumentationLibrary(b[i]), extractInstrumentationLibrary(b[j]))
+	return diff <= 0
+}
+
+func (b byOTLPTypes) Swap(i, j int) {
+	b[i], b[j] = b[j], b[i]
+}
+
+// V2SpansToInternalTraces translates Zipkin v2 spans into internal trace data.
```

Comment:
It doesn't appear that this method shares any implementation details with the Zipkin v1, can we keep them somewhat unified? I had just merged them together recently https://github.com/open-telemetry/opentelemetry-collector/pull/1002